pkg/receive/multitsdb.go

```diff
@@ -116,14 +121,57 @@ func NewMultiTSDB(
 
 func (t *MultiTSDB) GetTenants() []string {
 	t.mtx.RLock()
+	defer t.mtx.RUnlock()
 	tenants := make([]string, 0, len(t.tenants))
 	for tname := range t.tenants {
 		tenants = append(tenants, tname)
 	}
-	defer t.mtx.RUnlock()
```

Comment:
aha, found a bug cc @yuchen-db, you should defer Unlock right after

Reply:
thanks